app/src/main/java/ch/sdp/vibester/activity/ProfileActivity.kt

```diff
@@ -172,21 +215,43 @@ class ProfileActivity : AppCompatActivity() {
                 dataGetter.setFieldValue(FireBaseAuthenticator().getCurrUID(), "username",  input.text.toString())
             }
         }
-
         builder.setNegativeButton("Cancel") { dialog, _ -> dialog.cancel() }
         builder.show()
     }
 
     /**
-     * A function that displays the dialog.
-     * @param id: The id of the user to be shown.
-     * @param name: The name of the user to be shown.
+     * A function shows an image change dialog.
+     * @param title title of the dialog
      */
-    private fun showGeneralDialog(id: Int, name: String) {
-        val title = "Create $name"
-        val hint = "Enter new $name"
 
-        showDialog(title, hint, 0, id, name)
+    private fun showImageChangeDialog(title: String) {
+        val builder: AlertDialog.Builder = AlertDialog.Builder(this)
+        builder.setTitle(title)
+
+        builder.setPositiveButton("Yes") { _, _ ->
+            dataGetter.getCurrentUser()?.let { updateImage(it.uid) }
+        }
+
+        builder.setNegativeButton("No") { dialog, _ -> dialog.cancel() }
+        builder.show()

```

Comment:
Usually, builders have functions that return the instance on which the functions are applied to, so you can chain commands. It would be better if you could use that functionality here, i.e builder.fnc1(arg1).fnc2(arg2).show(), if possible.

Reply:
Yeah, but I think this will make it less readable :)

Reply:
You could chain them as such:
builder.fnc1(arg1)
           .fnc2(arg2)
           .fnc3(arg3)
           .show()
     
But yeah, currently it works as well, so all good

Reply:
Agreed, but the code works well regardless anyway